com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -1830,21 +1835,40 @@ internal void HandleSceneEvent(ulong clientId, FastBufferReader reader)
         /// Moves all NetworkObjects that don't have the <see cref="NetworkObject.DestroyWithScene"/> set to
         /// the "Do not destroy on load" scene.
         /// </summary>
-        private void MoveObjectsToDontDestroyOnLoad()
+        internal void MoveObjectsToDontDestroyOnLoad()
         {
             // Move ALL NetworkObjects to the temp scene
             var objectsToKeep = new HashSet<NetworkObject>(m_NetworkManager.SpawnManager.SpawnedObjectsList);
-
+            m_NetworkObjectDDOLActiveState.Clear();
             foreach (var sobj in objectsToKeep)
             {
-                if (!sobj.DestroyWithScene || (sobj.IsSceneObject != null && sobj.IsSceneObject.Value && sobj.gameObject.scene == DontDestroyOnLoadScene))
+                // Determine if we are already in the DDOL (i.e. moved by user)
+                bool isUserMovedNetworkObjectInDDOL = sobj.gameObject.scene == DontDestroyOnLoadScene;
+                if (!sobj.DestroyWithScene || isUserMovedNetworkObjectInDDOL)
                 {
                     // Only move objects with no parent as child objects will follow
                     if (sobj.gameObject.transform.parent == null)
                     {
-                        UnityEngine.Object.DontDestroyOnLoad(sobj.gameObject);
-                        // Since we are doing a scene transition, disable the GameObject until the next scene is loaded
-                        sobj.gameObject.SetActive(false);
+                        // If this is network object was moved into the DDOL by the user
+                        if (isUserMovedNetworkObjectInDDOL)
+                        {
+                            // Keep track of its active state in the m_NetworkObjectUserMovedDDOLActiveState list
+                            m_NetworkObjectUserMovedDDOLActiveState.Add(sobj.NetworkObjectId, sobj.gameObject.activeInHierarchy);
+
+                            // Since we are doing a scene transition, disable the GameObject until the next scene is loaded (LoadSceneMode.Single only)
+                            sobj.gameObject.SetActive(false);
+                        }
```

Comment:
just to note here: based on our internal conversation, we want to discover what would happen if we never touch object's active state at all — in other terms, what if we go back and delete `SetActive(false)` code we had there before.
if leaving objects as they are during scene transition wouldn't cause any issues internal to the SDK, we could simply not touch it and leave it up to user to decide how they want to handle objects ticking during scene transition.